### ISSUE TITLE
Shopping cart for multiple companies

### ISSUE THREAD

**Issue by sachinhub** [frappe/erpnext#9234]
In ERPNext we can create multiple 'Company' records in one site. We can also create Sales Orders for each of these companies. However it is not possible to create sales orders for multiple companies through 'Shopping cart'.

The 'Shopping cart' settings, currently allows cart to be created only for 1 company at a time. Is there any way we can enable cart for multiple companies? At the time of checkout, we can ask user to select the Company against which the sales order will be created?

**Comment by sachinhub** [comment]
If this looks a valid requirement, we can submit a pull request. Please provide your comments. I am thinking of adding one optional parameter - company in the place_order API.

If the parameter value is specified, Sales Order will be created against that company, otherwise the current logic will be followed.

**Comment by sunnyakaxd** [comment]
+1

**Comment by federojoavellaneda** [comment]
+1

**Comment by Devens** [comment]
+1


Thanking you,

With Regards,
Deven Shah
Safeworld Systems Pvt Ltd
1463/64, Khatraj Kalol Road,
Near Old Arvind Mill, Ta Kalol,
Dist Gandhinagar 382721
Gujarat India
Phone - 02764 281137/8

**Comment by nabinhait** [comment]
@sachinhub How the user will know the company to which the item belongs to? I think we have to link website items to company for this.

**Comment by sachinhub** [comment]
Hi @nabinhait Same item can be sold by multiple companies. What I was thinking is to simply add a company selector while placing the order in the shopping cart screen.

**Comment by nabinhait** [comment]
@sachinhub Yes, it's ok if the item is sold by multiple companies. But if the item is only sold by a company. Still we will show the company selection on shopping cart. And in that case, is the user select wrong company, it will create a Sales Order for wrong company. Also, in general the customer actually does not which company to select unless the customer communicates with you or a active customer since long time.

**Comment by puzzo-dev** [comment]
Whats the Update on this feature, i'm experiencing issues concerning this
